### ISSUE TITLE
Compiling Web version gives errors

### ISSUE THREAD

**Issue by BopItFreak** [sm64pc/sm64pc#305]
I tried to compile the web version using the nightly branch and the master branch. I even tried checking out to older commits as well. None of these seem to compile for the web version. I get various errors. Has anyone been able to compile it successfully? If so, what branch/commit did you use?

**Comment by coltongit** [comment]
Compiles fine for me with the latest nightly

**Comment by BopItFreak** [comment]
I get this issue @coltongit 
![image](https://user-images.githubusercontent.com/41522720/83921209-de2b8000-a74b-11ea-8f4d-24ceadbd0d9a.png)

**Comment by coltongit** [comment]
Make sure the emsdk is installed first

**Comment by BopItFreak** [comment]
It's installed.

**Comment by coltongit** [comment]
I'm honestly puzzled by this. Shouldn't ever error at this point.

**Comment by BopItFreak** [comment]
Ok, I deleted the whole folder and recompiled everything. I got this error now: 
![image](https://user-images.githubusercontent.com/41522720/83926103-e50bc000-a756-11ea-883d-5f81a8f20bf1.png)

How do I fix this?

**Comment by BopItFreak** [comment]
Alright, I edited some flags in the makefile to fix the memory issue, but now I got an error similar to the one I got before: 
![Q3c7K](https://user-images.githubusercontent.com/41522720/83928323-66665100-a75d-11ea-97fc-3bd52e2ac6b0.png)

Not sure what's going on. @coltongit

**Comment by BopItFreak** [comment]
Hmm, I guess the file doesn't exist? 
![image](https://user-images.githubusercontent.com/41522720/83929006-ccec6e80-a75f-11ea-9ded-c4a90aa525b1.png)

**Comment by BopItFreak** [comment]
Update: I was able to get it working using WSL.

**Comment by BopItFreak** [comment]
I've been able to compile the web version, but the gamepad api is not working at all. I don't understand why gamepads just don't work on the web version. Any ideas?

**Comment by Jan200101** [comment]
Make sure you are using a browser that supports the gamepad api and if your controller uses xinput

**Comment by BopItFreak** [comment]
@Jan200101 I tried it using Chrome and Firefox. Neither of them worked. Also a few people tried it as well and they couldn't get their controller working. My controller works on the regular version so I don't know why It doesn't work here. I even did some debugging in the console and I found that the web version detects the controller via the gamepad api, but for some reason isn't letting me assign keybinds and use it.

**Comment by Tikilou** [comment]
> @Jan200101 I tried it using Chrome and Firefox. Neither of them worked. Also a few people tried it as well and they couldn't get their controller working. My controller works on the regular version so I don't know why It doesn't work here. I even did some debugging in the console and I found that the web version detects the controller via the gamepad api, but for some reason isn't letting me assign keybinds and use it.

It could be a problem of groups/rights USB ? I'm on Manjaro Linux, and it works with plugdev and input group added for my user.
On Firefox, i've enabled Gamepad support in about:config too 
![Capture d’écran de 2020-06-15 16-49-31](https://user-images.githubusercontent.com/5641661/84671912-37539a80-af28-11ea-86d5-71c528b044b9.png)

**Comment by BopItFreak** [comment]
I tested my controller on a gamepad tester site. It worked perfectly. I
can't link my web version to this, but it would be nice if you compiled it
yourself to test.

**Comment by TheGreatestJannet** [comment]
what did you change to fix the memory error?

**Comment by BopItFreak** [comment]
@TheGreatestJannet I edited some memory thing in the makefile

**Comment by TheGreatestJannet** [comment]
yep you edit the TOTAL_MEMORY=25MB to fix the error. However it won't run in a browser with a different error currently.
